node/src/accountant/mod.rs

```diff
@@ -513,6 +540,74 @@ impl Accountant {
         }
     }
 
+    //for debugging only
+    fn calculate_debt_extremes(all_non_pending_payables: &[PayableAccount]) -> String {
+        if all_non_pending_payables.is_empty() {
+            "Payable scan found no debts".to_string()
+        } else {
+            struct PayableInfo {
+                balance: i64,
+                age: Duration,
+            }
+            let now = SystemTime::now();
+            let init = (
+                PayableInfo {
+                    balance: 0,
+                    age: Duration::ZERO,
+                },
+                PayableInfo {
+                    balance: 0,
+                    age: Duration::ZERO,
+                },
+            );
+            let (biggest, oldest) = all_non_pending_payables.iter().fold(init, |sofar, p| {
+                let (mut biggest, mut oldest) = sofar;
+                let p_age = now
+                    .duration_since(p.last_paid_timestamp)
+                    .expect("Payable time is corrupt");
+                if (p.balance > biggest.balance)
+                    || (p.balance == biggest.balance && p_age > biggest.age)
```

Comment:
A set of tests that demonstrated the necessity for this cross-dependency would make things clearer for the reader. Also, since this isn't a simple string in a debug! log, it should be tested anyway.